### ISSUE TITLE
oasis-node binaries made with GoReleaser via GitHub Actions are not reproducible

### ISSUE THREAD

**Issue by tjanez** [oasislabs/oasis-core#2571]
##### SUMMARY
When testing the release tooling I noticed that `oasis-node` builds are not reproducible, i.e. rebuilding the same git tag checkout does not produce the same `oasis-node` binary.

##### ISSUE TYPE
- Bug Report

##### OS / ENVIRONMENT
Go 1.13 on Ubuntu 18.04 LTS as [provided by GitHub Actions](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1804-lts).

##### ACTUAL RESULTS

```
[tadej@toronto The Quest]$ sha256sum oasis-node_20.1_linux_amd64*/oasis-node
db51791fa2c825e5d43ab87d6ff8bcc743368aeea4a016ea1d3fb851a10bc499  oasis-node_20.1_linux_amd64/oasis-node
c3ae733d5ae34d4959989b4535a98d9ba85bb9b9c29fb40a8a36fc548e1325f0  oasis-node_20.1_linux_amd64-REBUILD/oasis-node
```

##### EXPECTED RESULTS
Sha256sum of the rebuild of `oasis-node` from the same source tree should be equal to the released version.

**Comment by Yawning** [comment]
https://github.com/oasislabs/oasis-core/commit/60641ce41a9c2402f1b539375e1dd4e0eb45272d

```
arnhem :: /tmp % go tool buildid 20.1-release/oasis-node                       
qOH-_QEY3BGJ2Nn2bgJR/ws-c3GqZK-Cb5aG31TtO/JhNprcHDCpo5ywjw6DuT/mDmAV6jC5_nBoKuCiw3v
arnhem :: /tmp % go tool buildid 20.1-rebuild/oasis-node                       
RYljewfRd32AQmsKfEF5/TAwmr_XeZFaW2u3Jlf8P/JhNprcHDCpo5ywjw6DuT/x35OqgqT5Ht9ZHkApth1
```

**Comment by kostko** [comment]
Wasn't that included in latest Go (e.g., 1.13.6)?

**Comment by Yawning** [comment]
As far as I can tell the only differences in the binary are:
 * `.note.go.buildid`
 * `.note.gnu.build-id`

**Comment by tjanez** [comment]
It turned out this is a specific issue when performing builds via GoReleaser and using different git paths.
The ordinary `cd go && make oasis-node` builds are not affected.

Filled https://github.com/oasislabs/goreleaser/issues/1 and implemented a work-around in #2590.
